ui/src/components/connections/Connections.tsx

```diff
@@ -0,0 +1,179 @@
+import { Avatar, Card, Divider, Row, Space, Table, Tag, Tooltip, Typography } from "antd";
+import { ColumnsType } from "antd/es/table";
+import { useCallback, useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import { APIError } from "src/adapters/api";
+import { Connection, Credential, getConnections } from "src/adapters/api/connections";
+import { ReactComponent as IconUsers } from "src/assets/icons/users-01.svg";
+import { ConnectionsRowDropdown } from "src/components/connections/ConnectionsRowDropdown";
+import { ErrorResult } from "src/components/shared/ErrorResult";
+import { NoResults } from "src/components/shared/NoResults";
+import { SiderLayoutContent } from "src/components/shared/SiderLayoutContent";
+import { TableCard } from "src/components/shared/TableCard";
+import { useEnvContext } from "src/contexts/env";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+import { CONNECTIONS, QUERY_SEARCH_PARAM } from "src/utils/constants";
+import { AsyncTask, isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/types";
+
+export function Connections() {
+  const env = useEnvContext();
+  const [connections, setConnections] = useState<AsyncTask<Connection[], APIError>>({
+    status: "pending",
+  });
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const queryParam = searchParams.get(QUERY_SEARCH_PARAM);
+
+  const tableContents: ColumnsType<Connection> = [
+    {
+      dataIndex: "userID",
+      ellipsis: { showTitle: false },
+      key: "type",
+      render: (userID: Connection["userID"]) => (
+        <Tooltip placement="topLeft" title={userID}>
+          <Typography.Text strong>{userID.replace(/did:\w*:\w*:\w+:/, "")}</Typography.Text>
+        </Tooltip>
+      ),
+      sorter: ({ id: a }, { id: b }) => a.localeCompare(b),
+      title: "Identifier",
+    },
+    {
+      dataIndex: "credentials",
+      ellipsis: { showTitle: false },
+      key: "credentials",
+      render: (credentials: Credential[]) => (
+        <Space>
+          {[...credentials]
```

Comment:
`[...credentials]`?

Reply:
To perform the sort() operation and keep it immutable.

Reply:
Perform the sort() and keep the array immutable? As far as I'm aware we don't have any immutability rule enabled in the linter right? I mean, were you getting an error that made you add this?

I'd love to add [eslint-plugin-functional](https://github.com/eslint-functional/eslint-plugin-functional) to our `@iden3/eslint-config-react-ts` package if that's what you're insinuating! In that case, this would be required.

Reply:
@elias-garcia @cneuro do you agree on enforcing some functional rules and immutability among them?

Reply:
> I'd love to add [eslint-plugin-functional](https://github.com/eslint-functional/eslint-plugin-functional) to our @iden3/eslint-config-react-ts package if that's what you're insinuating! In that case, this would be required.

No, we won't add a full functional package to the iden3 config. I'm okay if you want to add rules related to immutability, like [this](https://github.com/jhusain/eslint-plugin-immutable) plugin

Reply:
Works for me. Can you take care of it @elias-garcia?